pkg/pillar/scripts/device-steps.sh

```diff
@@ -350,39 +350,38 @@ mkfifo /run/diag.pipe
 (while true; do cat; done) < /run/diag.pipe >/dev/console 2>&1 &
 $BINDIR/diag -f -o /run/diag.pipe runAsService &
 
-# Wait for having IP addresses for a few minutes
-# so that we are likely to have an address when we run ntp
-echo "$(date -Ins -u) Starting waitforaddr"
-$BINDIR/waitforaddr
+if [ ! -s $CONFIGDIR/device.cert.pem ]; then
+    # Wait for having IP addresses for a few minutes
+    # so that we are likely to have an address when we run ntp then create cert
+    echo "$(date -Ins -u) Starting waitforaddr"
+    $BINDIR/waitforaddr
 
-# Deposit any diag information from nim
-access_usb
+    # Deposit any diag information from nim
+    access_usb
 
-# We need to try our best to setup time *before* we generate the certifiacte.
-# Otherwise the cert may have start date in the future or in 1970
-echo "$(date -Ins -u) Check for NTP config"
-if [ -f /usr/sbin/ntpd ]; then
-    # '-p' means peer in some distros; pidfile in others
-    /usr/sbin/ntpd -q -n -p pool.ntp.org
-    # Run ntpd to keep it in sync.
-    /usr/sbin/ntpd -g -p pool.ntp.org
-else
-    echo "$(date -Ins -u) No ntpd"
-fi
+    # We need to try our best to setup time *before* we generate the certifiacte.
+    # Otherwise the cert may have start date in the future or in 1970
+    echo "$(date -Ins -u) Check for NTP config"
+    if [ -f /usr/sbin/ntpd ]; then
+        # '-p' means peer in some distros; pidfile in others
+        /usr/sbin/ntpd -q -n -p pool.ntp.org
+        # Run ntpd to keep it in sync.
+        /usr/sbin/ntpd -g -p pool.ntp.org
+        # Add ndpd to watchdog
+        touch "$WATCHDOG_PID/ntpd.pid"
+    else
+        echo "$(date -Ins -u) No ntpd"
+    fi
```

Comment:
Actually, we need it not only in case of generation of certificates. Without ntp and battery-backed clock ssl cannot verify certificates during connection.

Reply:
Correct. That is why I changed it to be inside the if statement on line 353.
Do you see something else which needs to be changed or commented?

Reply:
So, with your changes we will run ntp only in case of no device certificate. What I wrote: we need it in all cases for devices without battery (as a variant, you can check current year to decide the need for ntp).

Reply:
We have an ntp container doing that. We currently run two copies of ntpd!

Reply:
Ah, well, good point. However, let me check it with RPi

Reply:
Tested, it works, sorry for disturb